### ISSUE TITLE
error creating sandbox: 'error updating catalog: error decoding catalog: unexpected end of JSON input'

### ISSUE THREAD

**Issue by zarere** [datacharmer/dbdeployer#46]
**Describe the bug**
Not able to create sandbox

**To Reproduce**
1.Run
`
dbdeployer remote download 8.0.14
`
2.
`dbdeployer unpack 8.0.14.tar.xz `
3.
`dbdeployer deploy single 8.0.14`
it fails with the below error

```
Database installed in $HOME/sandboxes/msb_8_0_14
run 'dbdeployer usage single' for basic instructions'
error creating sandbox: 'error updating catalog: error decoding catalog: unexpected end of JSON input'
```

**Expected behavior**
A sandbox to be created 

**Environment:**
 - OS: Arch linux
 - dbdeployer 1.17.0
 - tarball full name (remote install or any version downloaded from 3-rd party sources)
 - tarball origin (https://dev.mysql.com/downloads/mysql/) 

**Hardware: (if applicable)**
 
`cat /etc/fstab && free -h && lscpu && sudo hdparm -I  /dev/sda |grep -i model && sudo hdparm -I  /dev/sdb |grep -i model`
# Static information about the filesystems.
# See fstab(5) for details.

# <file system> <dir> <type> <options> <dump> <pass>
# /dev/sda2
UUID=f5e21fab-e921-4e37-ac58-bf2ee4eefb2c	/         	ext4      	rw,noatime,barrier=0,data=ordered	0 1

# /dev/sda1
UUID=f30d9013-a27e-4195-a05c-4fc61f56764c	/boot     	ext4      	rw,relatime,stripe=4,data=ordered	0 2

# /dev/sda3
/dev/sda3           	none      	swap      	defaults,pri=-2	0 0

/dev/sdb5 /data         ext4            	       rw,noatime,barrier=0,data=ordered	0 2
              total        used        free      shared  buff/cache   available
Mem:           11Gi       3.1Gi       5.6Gi       345Mi       2.9Gi       8.1Gi
Swap:         511Mi          0B       511Mi
Architecture:        x86_64
CPU op-mode(s):      32-bit, 64-bit
Byte Order:          Little Endian
Address sizes:       36 bits physical, 48 bits virtual
CPU(s):              8
On-line CPU(s) list: 0-7
Thread(s) per core:  2
Core(s) per socket:  4
Socket(s):           1
NUMA node(s):        1
Vendor ID:           GenuineIntel
CPU family:          6
Model:               42
Model name:          Intel(R) Core(TM) i7-2760QM CPU @ 2.40GHz
Stepping:            7
CPU MHz:             797.367
CPU max MHz:         3500.0000
CPU min MHz:         800.0000
BogoMIPS:            4786.60
Virtualization:      VT-x
L1d cache:           32K
L1i cache:           32K
L2 cache:            256K
L3 cache:            6144K
NUMA node0 CPU(s):   0-7
Flags:               fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx rdtscp lm constant_tsc arch_perfmon pebs bts nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 cx16 xtpr pdcm pcid sse4_1 sse4_2 x2apic popcnt tsc_deadline_timer aes xsave avx lahf_lm epb pti ssbd ibrs ibpb stibp tpr_shadow vnmi flexpriority ept vpid xsaveopt dtherm ida arat pln pts flush_l1d
	Model Number:       CT120BX300SSD1                          
	Model Number:       Hitachi HTS541616J9SA00        

```
[martin@anarchy]: ~>$ locate /home/ json |grep deployer
/home/martin/.dbdeployer
/home/martin/.dbdeployer/sandboxes.json
[martin@anarchy]: ~>$ cat /home/martin/.dbdeployer/sandboxes.json
[martin@anarchy]: ~>$ stat /home/martin/.dbdeployer/sandboxes.json
  File: /home/martin/.dbdeployer/sandboxes.json
  Size: 0         	Blocks: 0          IO Block: 4096   regular empty file
Device: 802h/2050d	Inode: 3149784     Links: 1
Access: (0644/-rw-r--r--)  Uid: ( 1000/  martin)   Gid: (  985/   users)
Access dbdeployer --version
dbdeployer version 1.17.0
: 2018-03-28 14:51:50.754754304 +0300
Modify: 2018-04-20 11:34:33.440563175 +0300
Change: 2018-04-20 11:34:33.440563175 +0300
 Birth: -
[martin@anarchy]: ~>$ ls -lht /home/martin/.dbdeployer/sandboxes.json
-rw-r--r-- 1 martin users 0 Apr 20  2018 /home/martin/.dbdeployer/sandboxes.json
dbdeployer --version
dbdeployer version 1.17.0


```

**Comment by zarere** [comment]
` dbdeployer config show`
# Internal values:
{
 	"version": "1.17.0",
 	"sandbox-home": "$HOME/sandboxes",
 	"sandbox-binary": "$HOME/opt/mysql",
 	"use-sandbox-catalog": true,
 	"log-sb-operations": false,
 	"log-directory": "/home/martin/sandboxes/logs",
 	"master-slave-base-port": 11000,
 	"group-replication-base-port": 12000,
 	"group-replication-sp-base-port": 13000,
 	"fan-in-replication-base-port": 14000,
 	"all-masters-replication-base-port": 15000,
 	"multiple-base-port": 16000,
 	"group-port-delta": 125,
 	"mysqlx-port-delta": 10000,
 	"master-name": "master",
 	"master-abbr": "m",
 	"node-prefix": "node",
 	"slave-prefix": "slave",
 	"slave-abbr": "s",
 	"sandbox-prefix": "msb_",
 	"master-slave-prefix": "rsandbox_",
 	"group-prefix": "group_msb_",
 	"group-sp-prefix": "group_sp_msb_",
 	"multiple-prefix": "multi_msb_",
 	"fan-in-prefix": "fan_in_msb_",
 	"all-masters-prefix": "all_masters_msb_",
 	"reserved-ports": [
 		1186,
 		3306,
 		33060
 	],
 	"remote-repository": "https://raw.githubusercontent.com/datacharmer/mysql-docker-minimal/master/dbdata",
 	"remote-index-file": "available.json",
 	"timestamp": "Wed Jan 23 16:57:16 EET 2019"
 }

**Comment by zarere** [comment]
I deleted the empty file
/home/martin/.dbdeployer/sandboxes.json

and the error is gone now so closing the issue.

**Comment by datacharmer** [comment]
Thanks for your report.
I am able to reproduce it, and I think I understand what the problem could be.

First off, the sandbox is installed, despite what the error message says. The failure occurs while dbdeployer attempts to register the sandbox in the central catalog.

Try running `$HOME/sandboxes/msb_8_0_14/use` and it should access the database.

The error refers to an empty file that should be ignored or treated like an empty JSON structure, but apparently I made a mistake somewhere. The workaround that you have identified is effective. I will code a fix as soon as possible.

**Comment by datacharmer** [comment]
Fixed in release 1.17.1
